test/javascripts/acceptance/chat-user-menu-notifications-test.js

```diff
@@ -0,0 +1,252 @@
+import I18n from "I18n";
+import { test } from "qunit";
+
+import { click, visit } from "@ember/test-helpers";
+
+import {
+  acceptance,
+  exists,
+  query,
+  queryAll,
+  updateCurrentUser,
+} from "discourse/tests/helpers/qunit-helpers";
+
+import {
+  baseChatPretenders,
+  chatChannelPretender,
+} from "../helpers/chat-pretenders";
+
+acceptance(
+  "Discourse Chat - experiment user menu notifications - user cannot chat",
+  function (needs) {
+    needs.user({ has_chat_enabled: false });
+    needs.settings({ chat_enabled: false });
+
+    test("chat notifications tab is not displayed in user menu", async function (assert) {
+      await visit("/");
+      await click(".header-dropdown-toggle.current-user");
+
+      assert.notOk(
+        exists("#user-menu-button-chat-notifications"),
+        "button for chat notifications tab is not displayed"
+      );
+    });
+  }
+);
+
+acceptance(
+  "Discourse Chat - experimental user menu notifications ",
+  function (needs) {
+    needs.user({ redesigned_user_menu_enabled: true, has_chat_enabled: true });
+    needs.settings({ chat_enabled: true });
+
+    needs.pretender((server, helper) => {
+      baseChatPretenders(server, helper);
+      chatChannelPretender(server, helper);
+    });
+
+    test("chat notifications tab", async function (assert) {
+      updateCurrentUser({
+        grouped_unread_high_priority_notifications: {
+          29: 3, // chat mention notification type
+          31: 1, // chat invitation notification type
```

Comment:
```suggestion
          29: 3, // chat_mention notification type
          31: 1, // chat_invitation notification type
```

That way these lines will show up when grepping for the types names. Or if there are constants somewhere that we can that would be better.